client/js/components/procedure/admin/InstitutionTagManagement/TagListItem.vue

```diff
@@ -0,0 +1,313 @@
+
+      if (tagsInUse.length > 0) {
+        if (dpconfirm(Translator.trans('check.category_with_tags_in_use.delete', { category: name, count: tagsInUse.length, tags: tagsInUse.join(', ') }))) {
+          this.deleteCategoryAndTags()
+        }
+      } else if (tagsInUse.length === 0) {
+        if (dpconfirm(Translator.trans('check.category_with_tags_not_in_use.delete', { category: name }))) {
+          this.deleteCategoryAndTags()
+        }
+      }
+    },
+
+    confirmAndDeleteTag () {
+      const { isUsed, name } = this.item
+
+      if (isUsed) {
+        if (dpconfirm(Translator.trans('check.tag_is_used.delete'))) {
+          this.deleteTag()
+        }
+      } else if (!isUsed) {
+        if (dpconfirm(Translator.trans('check.tag.delete', { tag: name }))) {
+          this.deleteTag()
+        }
+      }
+    },
+
+    deleteCategoryAndTags () {
+      const { id, children, name } = this.item
+      const promises = [
+        this.deleteCategory(id),
+        ...children.map(tag => this.deleteTag(tag))
+      ]
+
+      Promise.allSettled(promises)
+        .then(() => {
+          dplan.notify.confirm(Translator.trans('confim.category_and_tags.deleted', { category: name }))
+          this.$emit('item:deleted', id)
+        })
+        .catch(error => {
+          console.error(error)
+        })
+    },
+
+    deleteCategory () {
+      const { id, name } = this.item
+
+      return this.deleteInstitutionTagCategory(id)
+        .then(() => {
+          dplan.notify.confirm(Translator.trans('confirm.category.deleted', { category: name }))
+        })
+        .catch(error => {
+          console.error(error)
+        })
+    },
+
+    deleteItem () {
+      const { type } = this.item
+      const isCategory = type === 'InstitutionTagCategory'
+      const isTag = type === 'InstitutionTag'
+
+      if (isCategory) {
+        this.handleCategoryDeletion()
+      } else if (isTag) {
+        this.confirmAndDeleteTag()
+      }
+    },
+
+    deleteTag (tag = this.item) {
+      return this.deleteInstitutionTag(tag.id)
+        .then(() => {
+          dplan.notify.confirm(Translator.trans('confirm.tag.deleted', { title: tag.name }))
+          this.$emit('tagIsRemoved')
+          this.$emit('item:deleted', tag)
+        })
+        .catch(error => {
+          console.error(error)
+        })
+    },
+
+    edit () {
+      this.isEditing = true
+    },
+
+    handleCategoryDeletion () {
+      const { children } = this.item
+      const hasTags = children.length > 0
+
+      if (!hasTags) {
+        this.confirmAndDeleteCategory()
+      } else {
+        this.confirmAndDeleteCategoryWithTags()
+      }
+    },
+
+    save () {
+      const { name, type } = this.item
+
+      if (name === this.name) {
+        this.isEditing = false
+        return
+      }
+
+      if (type === 'InstitutionTagCategory') {
+        this.saveCategory()
+      }
+
+      if (type === 'InstitutionTag') {
+        this.saveTag()
+      }
```

Comment:
If we wanna go fancy, we could do something like 
```suggestion
      this[`safe${type}`]()
```
and then just rename the methods

Reply:
As long as we don't use typescript, I prefer to keep it explicit, so we can see instantly which values `type` can have. What do you think?